FPM: refactor fpm_php_get_string_from_table() to better match usage

Girgias opened the pull request:
Pass in length of the key to improve the existance of the key check.

Feel free to close if you don't agree @bukka, but the returned ``char*`` is never used and passing in the length of the key allows performing the quick check of "do both string lengths match".

bukka commented on sapi/fpm/fpm/fpm_php.h:
```diff
@@ -3,6 +3,7 @@
 #ifndef FPM_PHP_H
 #define FPM_PHP_H 1
 
+#include <stdbool.h>
```
why do you need this? Isn't it already included in php.h ...?

Girgias replied:
Should be yes, just because I prefer to be explicit but can remove it.

bukka replied:
yes please, remove it.

bukka replied:
it doesn't make sense to be done only for a single change - it would need to be applied everywhere otherwise it's just inconsistent.

bukka commented on sapi/fpm/fpm/fpm_status.c:
```diff
@@ -172,7 +172,7 @@ int fpm_status_handle_request(void) /* {{{ */
 		/* full status ? */
 		_GET_str = ZSTR_INIT_LITERAL("_GET", 0);
-		full = (fpm_php_get_string_from_table(_GET_str, "full") != NULL);
+		full = fpm_php_is_key_in_table(_GET_str, "full", strlen("full"));
```
it's used below and looks better

Girgias replied:
Huh TIL about this macro

bukka commented on sapi/fpm/fpm/fpm_php.h:
```diff
@@ -41,6 +42,6 @@ void fpm_php_soft_quit(void);
 int fpm_php_apply_defines_ex(struct key_value_s *kv, int mode);
 int fpm_php_limit_extensions(char *path);
-char* fpm_php_get_string_from_table(zend_string *table, char *key);
+bool fpm_php_is_key_in_table(zend_string *table, char *key, size_t key_len);
```
```suggestion
bool fpm_php_is_key_in_table(zend_string *table, const char *key, size_t key_len);
```

Girgias commented:
Addressed the review comments, only other change was to move the declaration of the ``zend_string*`` inside the ``if`` branch as that's where it is only used.
